### ISSUE TITLE
datetimes not inserted with item creation

### ISSUE THREAD

**Issue by jisantuc** [azavea/franklin#724]
**Describe the bug**

https://github.com/azavea/franklin/pull/630 pulled out and indexed the datetime fields of items. That's great, but for new items, we don't actually insert those fields 🤦🏻‍♂️ (and we sensibly don't fall back to item properties). This can lead to surprising behavior, where you query for items that definitely match your datetime query, and you don't get any back.

**Expected behavior**
In import and item POST, those fields should be populated

**Additional context**
We could do this quick and dirty but I'd rather not -- instead, since we know things about datetime being required, we should do the ADT thing from https://github.com/azavea/stac4s/issues/120#issuecomment-809713114 and handle the insert accordingly

**Comment by pomadchin** [comment]
Franklin does not not properly fill `start_datetime`, `end_datetime` or `datetime` table fields, see [StacItemDao insert methods](https://github.com/azavea/franklin/blob/master/application/src/main/scala/com/azavea/franklin/database/StacItemDao.scala#L266).

TLDR; **All temporal queries are non functional right now.**

A motivating picture of the `collection_items` after the fresh catalog ingest with the most up to date Franklin:

![image](https://user-images.githubusercontent.com/4929546/117698789-145e5480-b192-11eb-90a2-4b0a0e22dbd8.png)
